### PR TITLE
[PORT] Faxes from TG

### PR DESCRIPTION
**Pull request by Bobbanz1**
## About The Pull Request
This PR ports the fax machine, alongside the various PRs associated with it. 

Ports the following PRs:
* https://github.com/tgstation/tgstation/pull/69083
* https://github.com/tgstation/tgstation/pull/69397
* https://github.com/tgstation/tgstation/pull/69554
* https://github.com/tgstation/tgstation/pull/69651
* https://github.com/tgstation/tgstation/pull/70072
* https://github.com/tgstation/tgstation/pull/71262

## To-Do

- [x] Write the changelog
- [x] Remove the code from my previous attempt at trying to figure out how to allow the faxing of photographs to Admins.
- [X] Fix the current issues with the Fax Machine associated? code
- - [X] Fax Manager TGUI bluescreens if there’s no fax machines in existence on the server. 
- - [X] Fax Machine TGUI crashes if there’s less then 2 Fax Machines in existence.
- [ ] Fix issues associated with the Fax Manager
- - [ ] Fax Manager randomly looses the stored fax IDs and the history of stuff sent to it. 
- - - [ ] Needs Re-Verification
- - [x] Faxes sent by Admins seem to initially be sent from Central Command but then get defaulted to " " which results in the fax missing a sender in the machines history
- - [ ] Paper code markdown not working on papers sent using Fax Manager, basic support available currently but doesn’t support input lines.
- - - [ ] Will be fixed upon completion and Porting of 90% of this PR: [#8113](https://github.com/BeeStation/BeeStation-Hornet/pull/8113)
- [x] Add Fax machines to the current maps in Rotation:
- - [x] Tycoon
- - [x] Aetherwhisp
- - [x] ~~Shrike?~~
- - [x] Eclipse
- - [x] Galactica
- - [x] Atlas
- - [x] Snake
- - [x] Hammerhead

## Why It's Good For The Game
Allows for closer IC interactions with Centcom/Admins through Faxes.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:twilightwanderer, Imaginos16, dragomagol, Jacquerel
add: Fax Machine
fix: Fixes faxes playing the "fax_contain" animation and makes it actually use the "fax_receive" animation when receiving.
imageadd: Resprites fax machines!
admin: fax messages and book printings are now logged in paper.log
admin: renaming fax machines is logged in game log
add: You can now send photos or trading cards through the fax machine.
add: You can now hack the wires of a fax machine to have it throw items harder or accept additional flat input items such as material sheets, ID cards, and pizza slices.
add: Ability to send and receive administrator fax messages.
admin: Panel for easy viewing and sending messages to faxes.
fix: Fixes a crash of TGUI fax panel when there is no fax in the network
fix: Fixes a crash of the Fax Manager panel when there is no fax machines in the network.
/:cl:

### CODE REVIEW

**Review by Karmic-Skink — CHANGES_REQUESTED** [review]
Update the CL to include the original PR author's handles.

**Comment by Bobbanz1** [comment]
> If you need any help, I can help you with implementation. But before you implement the port, I would advise you to look at the redesigned [TG paper](https://github.com/tgstation/tgstation/pull/68612), as well as the [Jolly fax variation](https://github.com/TaleStation/TaleStation).

I believe I was going to port the TG paper code system in this [PR](https://github.com/BeeStation/NSV13/pull/2023).

However I think the reason I ultimately choose not to do so was because of this, still open, [issue](https://github.com/tgstation/tgstation/issues/69261). The issue in question reported the experience of lag when writing on the paper.

**Comment by Karmic-Skink** [comment]
Can this be integrated into requests consoles at all?

**Comment by Bobbanz1** [comment]
@Karmic-Skink Could you explain what the Request Console is?

**Comment by Bobbanz1** [comment]
@twilightwanderer 
Hey there, I'm currently having an issue with both the Fax Machine TGUI and the Fax Manager TGUI, the issue is two folded (think that's the correct word for it) 

The first part is that if someone were to deconstruct every fax machine except for one lone fax machine on the server, then the Fax Machine TGUI will bluescreen. 

The second part is that if someone were to deconstruct EVERY fax machine on the server, then the Fax Manager TGUI will bluescreen until a new fax machine is built or spawned in.

I don't actually know enough about TGUI code to implement some sort of fix for the problem and was hoping you might have an idea on how to fix these two issues?

**Comment by Bobbanz1** [comment]
> Update the CL to include the original PR author's handles.

This has been done

**Comment by Karmic-Skink** [comment]
> @Karmic-Skink Could you explain what the Request Console is? Is it something that's related to Cargo or is it some sort of Admin panel thing?

image

/obj/machinery/requests_console

**Comment by Bobbanz1** [comment]
> > @Karmic-Skink Could you explain what the Request Console is? Is it something that's related to Cargo or is it some sort of Admin panel thing?
> 
> image
> 
> /obj/machinery/requests_console

Ah that one....No I don't believe it could be integrated into that console.

**Comment by Karmic-Skink** [comment]
> > > @Karmic-Skink Could you explain what the Request Console is? Is it something that's related to Cargo or is it some sort of Admin panel thing?
> > 
> > 
> > image
> > /obj/machinery/requests_console
> 
> Ah that one. Short Answer: No I don't believe it could be integrated into that console.
> 
> Detailed explanation: It would require a full reworking of the Request Console UI, to the point where it would be so far out of the scope of what this PR is trying to do that it would be an entire separate PR that would also have to be ported to Bee.
> 
> Currently the Fax Machine runs on TGUI code, the request console runs using HTML code, in order to even try figuring out how such an integration would even theoretically be done, I'd have to upgrade the Request Console's UI to TGUI code, something I'm not that good at and porting some codebases request console where it has been upgraded would be an entirely separate PR worth of Content.

It makes far more sense to integrate the function of these devices rather than having additional machines in rooms that already contain machines designed to facilitate communications outside of direct radio comms.

**Comment by Bobbanz1** [comment]
> > > > @Karmic-Skink Could you explain what the Request Console is? Is it something that's related to Cargo or is it some sort of Admin panel thing?
> > > 
> > > 
> > > image
> > > /obj/machinery/requests_console
> > 
> > 
> > Ah that one. Short Answer: No I don't believe it could be integrated into that console.
> > Detailed explanation: It would require a full reworking of the Request Console UI, to the point where it would be so far out of the scope of what this PR is trying to do that it would be an entire separate PR that would also have to be ported to Bee.
> > Currently the Fax Machine runs on TGUI code, the request console runs using HTML code, in order to even try figuring out how such an integration would even theoretically be done, I'd have to upgrade the Request Console's UI to TGUI code, something I'm not that good at and porting some codebases request console where it has been upgraded would be an entirely separate PR worth of Content.
> 
> It makes far more sense to integrate the function of these devices rather than having additional machines in rooms that already contain machines designed to facilitate communications outside of direct radio comms.
> 
> So effectively a -1 from me, as it is just adding additional clutter to already complicated areas.

Could you clarify what areas are complicated?

**Review by covertcorvid — APPROVED** [review]

**Review by Karmic-Skink — CHANGES_REQUESTED** [review]
Several map design errors.

**Review by covertcorvid — CHANGES_REQUESTED** [review]
Apparently not being logged in breaks the fax manager

File: _maps/map_files/Shrike/Shrike1.dmm [hunk 1]
```diff
@@ -10048,10 +10060,13 @@
 	},
 /obj/machinery/power/apc/auto_name/north,
 /obj/structure/table/reinforced,
-/obj/item/book/manual/wiki/sop/command,
 /obj/structure/cable/white{
 	icon_state = "4-8"
 	},
+/obj/machinery/fax{
```
[the remark this answers is not in the document]
Comment (Bobbanz1) [reply _maps/map_files/Shrike/Shrike1.dmm:10066]: I understand the issues but I really have to ask…How does a plant which isn’t even solid, restrict access around doors?

[the remark this answers is not in the document]
Comment (Karmic-Skink) [reply _maps/map_files/Shrike/Shrike1.dmm:10066]: Because you put a tripping hazard on the floor.

**Comment by Karmic-Skink** [comment]
Could you please just revert all Shrike changes and I'll manually add them if this PR is merged. Thanks.

**Comment by Bobbanz1** [comment]
> Apparently not being logged in breaks the fax manager

Issue has potentially been solved as I haven’t received any more runtime errors from Dusklight, Altangy, Ta or any other admins since my last commit. 

However there are still issues with the PR that needs fixing and I’m positive that there’s probably going to be more runtime errors cropping up eventually, is there a tag that keeps this PR from being merged but still kept as Test Merged?

**Comment by Bobbanz1** [comment]
> This pull request has conflicts, please resolve those before we can evaluate the pull request.

DAMN IT NOT AGAIN

**Review by covertcorvid — APPROVED** [review]

**Comment by Bobbanz1** [comment]
It’s always the fucking hammerhead to have merge issues!

**Comment by covertcorvid** [comment]
> > Apparently not being logged in breaks the fax manager
> 
> Issue has potentially been solved as I haven’t received any more runtime errors from Dusklight, Altangy, Ta or any other admins since my last commit.
> 
> However there are still issues with the PR that needs fixing and I’m positive that there’s probably going to be more runtime errors cropping up eventually, is there a tag that keeps this PR from being merged but still kept as Test Merged?

Are these issues fixed?

**Comment by Bobbanz1** [comment]
> > > Apparently not being logged in breaks the fax manager
> > 
> > 
> > Issue has potentially been solved as I haven’t received any more runtime errors from Dusklight, Altangy, Ta or any other admins since my last commit.
> > However there are still issues with the PR that needs fixing and I’m positive that there’s probably going to be more runtime errors cropping up eventually, is there a tag that keeps this PR from being merged but still kept as Test Merged?
> 
> Are these issues fixed?

I have not received any more issues from those admins howeverat the same time...this PR hasn't been test-merged in months and I'm still waiting on Beestation to get around to looking at the PR I made that updates the Papercode which would allow the admin interface to properly work.

**Comment by Bobbanz1** [comment]
After a few days of thought and a coin flip , I’ve come to a conclusion that I’ve reached the limit of what I can do for NSV, as such I’m going to slowly start pulling away from creating PRs and porting stuff for NSV13, as such I’m closing this PR.

**Comment by covertcorvid** [comment]
Sorry to hear that. I'd be willing to take this after re-verifying without having it ported to Bee if you were interested in that, but understandable if not.

**Comment by Bobbanz1** [comment]
> Sorry to hear that. I'd be willing to take this after re-verifying without having it ported to Bee if you were interested in that, but understandable if not.

I'll probably work on it again in the future once I've managed to regain some sort of future stability compared to the current crisis I'm facing, this PR is...well it works but it's also broken at the same time, it's missing specific PRs I was porting to Beestation to inevitably port downstream which would have made this PR work as it was supposed to work. 

Current issues are the following
* The formatting just does not work at all, you can't make fields nor do any of the formatting you can do on a regular piece of paper in-game.

**Comment by covertcorvid** [comment]
Ah yeah, I remember that part not working. Best of luck with your situation.

**Pull request closed without merging**